### ISSUE TITLE
grapheneos fdroid "No app store installed"

### ISSUE THREAD

**Issue by mattfbacon** [stevesoltys/backup#816]
After installing F-Droid I still get the message "no app store installed" when I try to manually install apps that couldn't be automatically restored.

**Comment by grote** [comment]
May be a Graphene specific issue. What app did installed these apps previously, when you did the backup? The same F-Droid?

Could you leave restore right after the message "no app store installed" appeared, then in the main screen, go to expert settings and export logcat from there? I'd be interested in the end of that log.

The code that creates the app store intent is here:  https://github.com/seedvault-app/seedvault/blob/7c8575e17a71497a7351ad77f0605c553194ddb2/app/src/main/java/com/stevesoltys/seedvault/restore/install/InstallIntentCreator.kt#L15

It uses a `market://details?id=$packageName` URI which all stores including F-Droid normally understand. If the installer of the app is known *and* the installer app is installed, we restrict the intent to that. There may be a bug in this logic, but lets wait for your log.

**Comment by mattfbacon** [comment]
Yeah so they probably had org.fdroid.basic as the installation source because I was using CalyxOS and now I have org.fdroid.fdroid. But there should be a case for when the old installation source is no longer present that allows the normal behaviour of choosing any installer.

**Comment by grote** [comment]
I guess `org.fdroid.basic` is not installed then? Can you try with it installed, if it fixes the issue?

The logic should not restrict the intent to basic, if it isn't installed: https://github.com/seedvault-app/seedvault/blob/7c8575e17a71497a7351ad77f0605c553194ddb2/app/src/main/java/com/stevesoltys/seedvault/restore/install/InstallIntentCreator.kt#L46-L60

So F-Droid should be able to pick up the intent we send.

**Comment by grote** [comment]
F-Droid Basic installer matching first entered in https://github.com/seedvault-app/seedvault/commit/18013c03796aa2fd0fe857279516d5382df171fa which first appeared in 14-4.0 which is newer than 14-3.3 what you use. So that means the package of the intent should not get restricted and any app store can pick it up.

If you install aurora and/or F-Droid basic, is any app getting opened?

**Comment by mattfbacon** [comment]
Yes, with F-Droid Basic installed it did work.

**Comment by grote** [comment]
That is weird. This sounds like it had restricted it to basic even though it isn't installed. Or normal F-Droid wasn't reacting to the intent?

**Comment by mattfbacon** [comment]
I think it was an corner case from the fact that I had just installed normal F-Droid during the backup restore, F-Droid Basic itself was installed by the restore and I uninstalled it manually after, and the packages to be restored were marked as installed by F-Droid Basic. I can't reproduce the issue anymore and I also tested with some testing code to invoke the market URL intent and everything worked normally.
